parser/charset/encoding.go

```diff
@@ -29,6 +29,16 @@ func IsSupportedEncoding(charset string) bool {
 	return ok
 }
 
+// FindEncodingUTF8AsNoop finds the encoding according to charset
+// except that utf-8 is treated as binary encoding.
+func FindEncodingUTF8AsNoop(charset string) Encoding {
```

Comment:
1. Please add a test case for this function.
2. The comment is still confusing to me at least, what is the case when "utf-8 is treated as binary encoding"?

Reply:
1. I think the logic here is simple enough...?
2. Binary encoding is a noop encoding. Different from utf-8 encoding, it means all the methods are trivial, including `Transform()`, `IsValid()` and others. The cost should be O(1) for these operations. I am trying to avoid string validation(before parsing or writing result to client) to see if there is any improvement.

Reply:
> Binary encoding is a noop encoding. Different from utf-8 encoding, it means all the methods are trivial, including Transform(), IsValid() and others. The cost should be O(1) for these operations. I am trying to avoid string validation(before parsing or writing result to client) to see if there is any improvement.

Thanks! I think this can be added to the comment in some way...

@Yui-Song Could you try to validate the result using your benchmark?

Reply:
> > Binary encoding is a noop encoding. Different from utf-8 encoding, it means all the methods are trivial, including Transform(), IsValid() and others. The cost should be O(1) for these operations. I am trying to avoid string validation(before parsing or writing result to client) to see if there is any improvement.
> 
> Thanks! I think this can be added to the comment in some way...
> 
> @Yui-Song Could you try to validate the result using your benchmark?
I have told @tangenta how to validate it easily with Benchbot, a platform provided by QA team to do benchmarks. All the benchmarks the Perf Team run daily could be run with it.

Reply:
@Yui-Song  For the benchmark today, could you share the profiling files for both https://github.com/pingcap/tidb/commit/755553660e6036ead13d1f0aa271eaf59d939fbd and this commit?